XRTK.SDK/Packages/com.xrtk.sdk/Features/UX/Scripts/Editor/SceneHelpers.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEditor;
+using UnityEngine;
+
+namespace XRTK.SDK.UX.Utilities
+{
+    public class SceneHelpers
+    {
+        /// <summary>
+        /// Simple scene helper to create the beginnings of a scene, creating the scene root and a floor.
+        /// </summary>
+        [MenuItem("Mixed Reality Toolkit/Tools/SDK/Create Floor", false, 1)]
+        public static void CreateFloor()
```

Comment:
idk I feel like this should be something we're doing in the camera rig setup phase

Reply:
I'm also not sure this should be an SDK component if we're doing editor stuff

Reply:
It's an optional helper script that isn't part of the core functionality.  The SDK is intended to extend the usability of the core framework and provide UX / Usability features.  This fits into that category. I see no reason for this to be in core as "advanced" users will have their own way of doing things, but for a beginner (or tester) it is essential as a learning tool.

And no, we discussed setting up the rest of the scene previously and this is not part of the camera rig.

Reply:
I'm still apprehensive about this. I mean all it's really doing is creating a root object and a plane and renaming it.
Wouldn't people just do this anyway?

Also I'm not a fan of having a root `Scene Object` in my projects, and it's just a personal preference.

Idk, I just don't see the real benefit here, and would rather see something like this be part of the initial scene setup in the core.

Reply:
So, I am going to jump in here and provide my point of view. I see your point Stephen. For me, this utility classifies as "Dev Tools" or "Prototyping Tools". It's not something that can be considered a core piece of XRTK or a UX feature. This is simply for quickly getting started protoyping something.

That said, NOT ONCE have I created a new Unity project with XRTK that targets VR and have't created a floor / ground. I think that is what Simon is solving here. It's just a super simple dev utility that saves a few clicks.

"Idk, I just don't see the real benefit here, and would rather see something like this be part of the initial scene setup in the core."
-> That's what I mean. What you are describing is an actual solution people would use to get their projects done. Simons tool is just somethign you will delete/drop once you start fleshing out things. And let's be honest here, no one is gonig to use a simple plain in a final application for the ground.
-> That also said, I don't see XRTK having to care about setting up my scene and adding automatically any sort of geometry to it. Because the looks of a scene is always something that is very specific to a project.

Reply:
I think the floor should be part of the camera rig and we should give the configuration options there, and also offering it as this drop down menu item as well if we want.

Reply:
One other thing to note is that there might be a conflict with the Boundary system's floor and this one we're creating?

I'm also curious how we can also utilize the spatial awareness information about where the floor is too, when making any final adjustments to the floor offset.

Reply:
I don't want to have a GameObject container for the scene content when we can just adjust the camera rig universally for any platform type.

Reply:
I don't know. I don't see a floor having anything to do with a camera rig to be honest. But that's my personal opinion of course.

Reply:
Gotta be careful here that XRTK does not "overshoot" its responsibilities.

Reply:
Like in MRTK where they sometimes just do too much handholding IMO

Reply:
Well there are multiple ways to do this. 

You can have it so your rig is centered in the world, and everything moves around it.
Or you can have your player move through the world.

Depends on how you want to calculate the math for physics and other things.

Then there's the floating point precision problems.

Reply:
I just think that the few steps this script is doing is too obvious a thing that Unity wouldn't add to the editor themselves.

In fact in the test utilities we have some classes that do this kinda stuff like:

```
EditorSceneManager.NewScene(NewSceneSetup.DefaultGameObjects, NewSceneMode.Single);
```

Reply:
Why can't that just be an extension method in the core?

Reply:
If you feel that strongly we can move it to core. But this is intended to be a very simple helper script for devs prototyping or a quick start.  if you don't want to use it fine, but a lot of people will.
Can't believe we have such a debate over a simple helper script.

Reply:
I'm just saying that I think unity already provides some interesting built in functionality, let's extend on that and use it for our unit testing if needed.